### PR TITLE
Proposal:Keadm Tool Enhancement

### LINKED ISSUE #5563: Keadm Tool Enhancement
**What would you like to be added/modified**:Add New Flags to Keadm for Custom Configuration Support for EdgeCore installation

**Why is this needed**:We would like to support specifying parameters using `--set` or directly using an existing local configuration file to achieve full parameter configuration and meet the users' requirements.

### PR DESCRIPTION
**Pull request by HT0403**
**What type of PR is this?**
/kind design

**What this PR does / why we need it**:
Keadm(KubeEdge installation tool) now only supports configuring a subset of parameters during EdgeCore installation. We would like to support specifying parameters using --set or directly using an existing local configuration file to achieve full parameter configuration and meet the users' requirements.
**Which issue(s) this PR fixes**:

Fixes #5563 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by fisherxu — COMMENTED** [review]
/assign @WillardHu

**Comment by WillardHu** [comment]
Add some design contents about how to add a flag `set` to the keadm and applied to the edgecore config.

File: docs/proposals/Keadm Tool Enhancement.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+We would like to support specifying parameters using `--set` or directly using an existing local configuration file to achieve full parameter configuration and meet the users' requirements. 
+
+## Goals
+
+- Use '--set' or local configuration file to set parameters
+
+## Design Details
+
+### **The frame design drawing is as follows**：
```
Comment (WillardHu) [inline docs/proposals/Keadm Tool Enhancement.md:25]:
```markdown
### The frame design drawing is as follows
```

File: docs/proposals/Keadm Tool Enhancement.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+   - `key[...].key[M]` - The key indicates a specific field within an element of the array.
+   - `key[...].key[M].variable1` - The key indicates a variable within a specific field of an element in the array.
+5. **Further Split** - The key is further split based on whether there is an index position or member variables.
+6. **Use `reflect.ValueOf`** - Reflection is used to obtain the reflection value of the incoming structure pointer, then the `Elem` method is utilized to recursively find the reflection value of the field that needs modification.
+7. **Key and Structure Type Check (`key && tStruct`)** - It determines if the key matches the fields in the structure.
+8. **Modify Config** - The `Config` structure is modified according to the parsed content.
+9. **End of Keys** - All keys have been processed.
+10. **End** - The process ends.
+
+ ###  **Design contents about how to add a flag `set` **
```
Comment (WillardHu) [inline docs/proposals/Keadm Tool Enhancement.md:46]:
```markdown
### Design contents about how to add a flag `set`
```

File: docs/proposals/Keadm Tool Enhancement.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+```go
```
Comment (WillardHu) [inline docs/proposals/Keadm Tool Enhancement.md:48]: 1. You need add a flag define in the AddJoinOtherFlags() function, this function is both in file join_others.go and join_windows.go, you can refer to https://github.com/kubeedge/kubeedge/blob/7a81f8a3e603510b4d9b7dfedf6203c3cdfa83fd/keadm/cmd/keadm/app/cmd/cloud/init.go#L107
2. You need parse set slice values to edgeCoreConfig instance after it was initialized in the createEdgeConfigFiles() function this function is both in file join_others.go and join_windows.go too, you can refer to https://github.com/kubeedge/kubeedge/blob/7a81f8a3e603510b4d9b7dfedf6203c3cdfa83fd/keadm/cmd/keadm/app/cmd/edge/join_others.go#L168

**Review by WillardHu — CHANGES_REQUESTED** [review]

File: docs/proposals/Keadm Tool Enhancement.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+   - `key[...].key[M]` - The key indicates a specific field within an element of the array.
+   - `key[...].key[M].variable1` - The key indicates a variable within a specific field of an element in the array.
+5. **Further Split** - The key is further split based on whether there is an index position or member variables.
+6. **Use `reflect.ValueOf`** - Reflection is used to obtain the reflection value of the incoming structure pointer, then the `Elem` method is utilized to recursively find the reflection value of the field that needs modification.
+7. **Key and Structure Type Check (`key && tStruct`)** - It determines if the key matches the fields in the structure.
+8. **Modify Config** - The `Config` structure is modified according to the parsed content.
+9. **End of Keys** - All keys have been processed.
+10. **End** - The process ends.
+
+ ###  Design contents about how to add a flag `set`
```
Comment (WillardHu) [inline docs/proposals/Keadm Tool Enhancement.md:46]: Add some examples for `keadm join --set`

**Review by WillardHu — CHANGES_REQUESTED** [review]

**Review by WillardHu — APPROVED** [review]

**Comment by WillardHu** [comment]
Sorry, I approve my request changes, but the label approved have been added... 😂

**Comment by fisherxu** [comment]
> Sorry, I approve my request changes, but the label approved have been added... 😂

All right, if there's any issue, just fix it in follow up PRs :)

**Pull request merged**